src/java/org/apache/cassandra/service/consensus/migration/ConsensusRequestRouter.java

```diff
@@ -85,11 +87,17 @@ ConsensusRoutingDecision decisionFor(TransactionalMode transactionalMode)
 
     private static TableMetadata metadata(ClusterMetadata cm, String keyspace, String table)
     {
-        KeyspaceMetadata ksm = cm.schema.getKeyspaceMetadata(keyspace);
-        TableMetadata tbm = ksm != null ? ksm.getTableOrViewNullable(table) : null;
-
+        Optional<KeyspaceMetadata> ksm = cm.schema.maybeGetKeyspaceMetadata(keyspace);
+        if (ksm.isEmpty())
+        {
+            KeyspaceMetadata ksm2 = Schema.instance.getKeyspaceMetadata(keyspace);
+            if (ksm2 == null)
```

Comment:
CAS is supported on non-distributed keyspace (ya really) so this is all about preserving support for it. Since it's not distributed and Accord doesn't support non-distributed keyspaces this is letting Paxos continue to provide that support.

There was a bug where we could try to route to keyspaces that are not distributed and we would not return the right error here so this is differentiating between when it's an error and the keyspace doesn't exist and when it's just that it's not a distributed keyspace. I tried to standardize on this `metadata` and `getTableMetadata` method so we consistently perform this check.

Reply:
Also Paxos racing is acceptable because Paxos (and even non-SERIAL) can misroute. The ClusterMetadata check itself is inherently racy and could miss the fact that other nodes have adopted this table as an Accord table.

I'll add a few more checks to confirm the keyspace we find is actually local, but they don't eliminate this race and we can still pick paxos on the coordinator when the `ClusterMetadata` has been updated to claim it's an Accord table.